### PR TITLE
Search Indexing Connector

### PR DESCRIPTION
**Pull request by danielaotelea**
This is a draft PR. The implementation is not finished. I would like to check if the current implementation follows the pattern used for the OMRS connectors.
The purpose of this connector is to provide an efficient way to find entities by type from the cohort.
The configuration of the connector should provide the connection to Elastic search and the OM types used in for Enterprise OMRS topic events (the processing of these events should be based on the types: for PoC we choose Asset, Schema Elements, Glossary Terms).
The first part of the search indexing connector is part of the OMRS layer, the actual implementation of it is adapters - repository services.

This is an example of request body for connector config: {{server-url}}/open-metadata/admin-services/users/{{user-id}}/servers/{{server-id}}/search-indexing/configuration
{
    "class":"SearchIndexingConfig",
    "description":"Example Search Indexing Solution",
    "searchIndexingConnection":{
        "class":"Connection",
        "displayName":"Search Indexing Connection",
        "description":"Used for storing and fetching metadata in the Open Metadata format",
        "connectorType":{
            "class":"ConnectorType",
            "connectorProviderClassName":"org.odpi.openmetadata.adapters.repositoryservices.searchindexing.repositoryconnector.SearchIndexingRepositoryConnectorProvider"
        },
        "endpoint":{
        	"class": "Endpoint",
        	"address": "{{search-host}}:{{search-port}}"
        },
        "configurationProperties":{
            "storageBackend":"cassandra",
            "indexSearchBackend":"elastic-search"
        }
    }
}

Signed-off-by: Daniela Otelea <daniela-valentina.otelea@ing.com>

### CODE REVIEW

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--SPDX-License-Identifier: Apache-2.0-->
+<!--Copyright Contributors to the ODPi Egeria project.-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: There's a new section we need in all poms now - look for the `<scm>` tag. This is (apparently) necessary for our release publishing process. You can cut/paste from another pom

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    <parent>
+        <artifactId>open-metadata-implementation</artifactId>
+        <groupId>org.odpi.egeria</groupId>
+        <version>1.5-SNAPSHOT</version>
+        <relativePath>../../../../pom.xml</relativePath>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: The parent should normally be the default - ie one level up (..) -

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+<!--Copyright Contributors to the ODPi Egeria project.-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <parent>
+        <artifactId>open-metadata-implementation</artifactId>
+        <groupId>org.odpi.egeria</groupId>
+        <version>1.5-SNAPSHOT</version>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: master is now at 1.6-SNAPSHOT

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+        <dependency>
+            <groupId>org.odpi.egeria</groupId>
+            <artifactId>open-connector-framework</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.elasticsearch.client</groupId>
+            <artifactId>elasticsearch-rest-high-level-client</artifactId>
+            <version>7.5.1</version>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: A version should not be specified here - we should consolidate in the top level pom in the `<dependencyManagement>` section

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpcore</artifactId>
+            <version>4.4.12</version>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: remove version - use top level pom `<dependencyManagement>`

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.elasticsearch.client</groupId>
+            <artifactId>elasticsearch-rest-client</artifactId>
+            <version>7.5.1</version>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: remove version - use top level pom `<dependencyManagement>`

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.elasticsearch</groupId>
+            <artifactId>elasticsearch</artifactId>
+            <version>7.5.1</version>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: remove version - use top level pom `<dependencyManagement>`

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.elasticsearch</groupId>
+            <artifactId>elasticsearch-core</artifactId>
+            <version>7.5.1</version>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: remove version - use top level pom `<dependencyManagement>`

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>2.8.6</version>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: remove version - use top level pom `<dependencyManagement>`

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+        <dependency>
+            <groupId>org.elasticsearch</groupId>
+            <artifactId>elasticsearch-core</artifactId>
+            <version>7.5.1</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: This is a new direct dependency (though we do already pick it up transitively) - is there a reason we need to use this - if so we should bring this up on a discussion call

**Comment by planetf1** [comment]
Is there an issue related to this proposal?

**Comment by danielaotelea** [comment]
> Is there an issue related to this proposal?

Hi, Nigel! We have discussed about these proposals on Amsterdam meeting from January (and in some workshops before that), but you are right. It will be useful to open a Github issue to present the purpose of this connector.

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml (hunk 2 above)
Comment (planetf1) [reply open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: The parent is still wrong in my view - why are we using open-metadata-implementation?

We should follow the pattern adaopted under open-metadata-archive-connectors where we actually have a pom (of type packaging=pom) for each level, so as you traverse each directory down we specify as a module in it's parent, and also refer to the parent from the child module. 

So we need this for search-indexing-connectors for example

Perhaps there's a better way - but if so needs explanation, proposal, and aim to be consistent?

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml (hunk 3 above)
Comment (planetf1) [reply open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: fixed

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml (hunk 10 above)
Comment (planetf1) [reply open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: I see gson is now removed

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml (hunk 1 above)
Comment (planetf1) [reply open-metadata-implementation/adapters/open-connectors/repository-services-connectors/search-indexing-connectors/pom.xml]: still need this section

**Pull request closed without merging**